### PR TITLE
Use role recipe for magnum-server

### PR DESCRIPTION
**Pull request by mmnelemane**
Change magnum barclamp to use role_<> recipe for simplified role for magnum-server . Depends on the actual barclamp being merged : https://github.com/crowbar/crowbar-openstack/pull/360

### CODE REVIEW

**Comment by vuntz** [comment]
I guess only the last commit is worth reviewing -- rest comes from the other PR.

**Comment by jsuchome** [comment]
For the last commit: we currently use only the simplest way, no checking for `element_states`. I think it would be better to do it in line with other role recipes.

**Comment by vuntz** [comment]
Can you rebase, and add a commit for https://github.com/crowbar/crowbar-openstack/pull/360/files#r65667609 ?

File: chef/cookbooks/magnum/recipes/sql.rb [hunk 1]
```diff
@@ -26,13 +26,14 @@
 
 crowbar_pacemaker_sync_mark "wait-magnum_database"
 
+is_cluster_founder = CrowbarPacemakerHelper.is_cluster_founder?(node)
```
Comment (jsuchome) [inline chef/cookbooks/magnum/recipes/sql.rb]: Maybe I'm totally wrong, but are we sure this is executed at the same time as in that `only_if` section?

Comment (mmnelemane) [inline chef/cookbooks/magnum/recipes/sql.rb]: @jsuchome: I actally expect it wo work on the similar lines as the `'ha_enabled'`flag inside `only_if`

Comment (jsuchome) [inline chef/cookbooks/magnum/recipes/sql.rb]: It's different kind of check, but I believe it should work, yes.

Comment (mmnelemane) [inline chef/cookbooks/magnum/recipes/sql.rb]: @jsuchome, @vuntz: any idea why the patch fails to apply. I verified that its up-to-date and rebased with master.

Comment (vuntz) [inline chef/cookbooks/magnum/recipes/sql.rb]: The package needs to be updated to have the other PR. Doing that right now; feel free to restart the job in 5 minutes.

**Comment by jsuchome** [comment]
+1

**Comment by aplanas** [comment]
after that: +1

**Comment by mmnelemane** [comment]
@aplanas : Done. Thanks

**Comment by jsuchome** [comment]
Still +1

**Pull request merged**